### ISSUE TITLE
Echo PL issue reports to question repo item

### ISSUE THREAD

**Issue by echuber2** [PrairieLearn/PrairieLearn#1502]
Besides the suggestion of a more robust issue tracker on PL itself #1438 , it seems useful to somehow echo notes about issue reports filed on PL itself to the corresponding repo item for a question. Not sure how best to implement that, but I'll mention it here as an idea. (Perhaps something like this was already being developed?)

**Comment by nwalters512** [comment]
What do you mean by "corresponding repo item"? As in, automatically file an issue on GitHub referencing the question? Would be great to have a little more detail about what you were thinking.

**Comment by echuber2** [comment]
Yeah, something like that. If not to file an issue each time, perhaps to have a single issue log in the question directory or something, if issues would be spammy. I'm not really sure what would make the most sense, but it's difficult to have two different issue trackers with different feature sets tracking somewhat overlapping sets of issues.

**Comment by echuber2** [comment]
The visibility of issues on PL itself seems obscured. You really have to look at a question page directly to know there's an issue. I bet a lot of staff keep to the configuration files and manage courses without exploring the question browser.

**Comment by nwalters512** [comment]
We've thought about adding conversational replies to issues, but now I'm not sure if that's the right way to go about it. I wouldn't think of a PL issue in the same way that I would a GitHub issue - think of them more as an error log with context? There might not be a one-to-one mapping between issues and corresponding problems, such as when 10 students report the same broken question. I'd be hesitant to automatically create issues and instead let people manually make issues in GH or their project management software of choice.

I think in general we do a good job of making issues visible - they appear in the navbar, on question rows on the questions page, next to each assessment, and on individual instructor question pages. Is there a particular context you want to see them in where they aren't visible?

**Comment by echuber2** [comment]
The specific screen I was thinking of was where you click (as Instructor) Assessments > [assessment name] > Questions tab. Then it shows QID, Topic, Tags, etc. but not Issues. However, going through the topmost navbar Questions tab, and filtering by the exam name, accomplishes basically the same purpose.

With respect to the automatic issue logging, I agree making GitHub issues programmatically from every PL issue would be a bad idea... Actually it probably also wouldn't work to have a text log file in the question directory on the repo getting automatically changed.

Perhaps what I'm really looking for is some kind of auto notification feature that would nag if an assessment is populated with items that have outstanding issues, or a way to subscribe and receive notification whenever a student reports an issue on watched items? It would also be helpful to be able to remark internally on PL issues, if not to respond to them conversationally. E.g., to mark "this student complaint is in error" or "this was already fixed" or "this is a confirmed issue; do not use this question until resolved", etc. It's not clear currently to other staff what clicking "resolved" means for a PL item, unless the staff want to exhaustively log their interactions with those issues elsewhere for recordkeeping.

**Comment by echuber2** [comment]
I guess the simplest way to flag issues categorically right now is for the staff to add additional issues to an item with issues, just to leave an internal comment. (Prof. Zilles seems to have done so occasionally. I've also felt compelled to do this.) This feels like a makeshift solution though...

**Comment by mwest1066** [comment]
While I don’t think we don’t want to a full issue discussion system (like GitHub’s), it would be awesome to include resolution information on issues that we could display to students (as in #910) and show this to students (as in #1438). May we would have two extra columns in the `issues` table:

* `resolution` — student-visible field that instructors can edit
* `instructor_notes` — notes that are only visible to instructors, and which instructors can edit

**Comment by mwest1066** [comment]
@echuber2 above you say:

“The specific screen I was thinking of was where you click (as Instructor) Assessments > [assessment name] > Questions tab. Then it shows QID, Topic, Tags, etc. but not Issues.”

Can you confirm that this is indeed the case? Issues are supposed to show up on that screen (as a red number badge next to question titles). If they are not there, would you mind filing a new issue report specifically about this bug? Thanks!

**Comment by echuber2** [comment]
@mwest1066 The badge only seems to appear for issues that were raised for the given question *during* the given assessment. That is, if the question already had outstanding issues from previous inclusions in other assessments, those are not signified by a badge on the overview page, although they can be seen after clicking through to the specific question's page. Is this what was intended?

**Comment by mwest1066** [comment]
@echuber2 the behavior you describe (only showing issues for questions during the assessment) is intended behavior. I’d be very happy to revisit this decision, though. If you think it would be better to show all issues on the badges, I’d appreciate hearing your thoughts about why that would be better.

**Comment by echuber2** [comment]
Well, the basic situation is the one where I discovered it: questions from a previous exam being included for a current exam, and not realizing there were outstanding issues raised on a given question until drilling down further. It seems like if there are outstanding issues on the same question, regardless of which session/assessment the issue originated in, the instructor should be able to see it as readily as possible.

**Comment by mwest1066** [comment]
That seems reasonable. I think the original thinking was that for an exam, you really only want to know whether there are issues actually on that exam. But maybe this is all just too sneaky, especially given that there is no direct indication that issues are only showing for this question on the current assessment, rather than all issues for this question. I’ll file a new issue for this.
